hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -617,9 +624,27 @@ long acquireLock() throws UnknownHostException, TException, InterruptedException
             Lists.newArrayList(lockComponent),
             System.getProperty("user.name"),
             InetAddress.getLocalHost().getHostName());
-    LockResponse lockResponse = metaClients.run(client -> client.lock(lockRequest));
-    AtomicReference<LockState> state = new AtomicReference<>(lockResponse.getState());
-    long lockId = lockResponse.getLockid();
+
+    // Only works in Hive 2 or later.
+    if (MetastoreVersion.min(MetastoreVersion.HIVE_2)) {
+      lockRequest.setAgentInfo(agentInfo);
+    }
+
+    Pair<Long, LockState> lockResult;
+    try {
+      lockResult = tryLock(lockRequest);
+    } catch (TException e) {
+      // Try to find the lock. Only works in Hive 2 or later.
+      if (MetastoreVersion.min(MetastoreVersion.HIVE_2)) {
+        lockResult = reTryLock(agentInfo, lockRequest);
+      } else {
+        LOG.info("Could not retry with HMSClient {}", MetastoreVersion.current(), e);
```

Comment:
Should this be a higher level log, like `warn`?

Reply:
The Exception is rethrown, and most likely logged later.
This log line is only here to highlight the  codepath used